packages/components/src/components/entities/EntityInsertPanel.tsx

```diff
@@ -175,44 +217,53 @@ export class SampleInsertPanelImpl extends React.Component<Props, StateProps> {
     }
 
     init(props: OwnProps, selectTab: boolean = false) {
-
-        const queryParams = props.location ? SampleInsertPanelImpl.getQueryParameters(props.location.query) : {
+        const queryParams = props.location ? EntityInsertPanelImpl.getQueryParameters(props.location.query) : {
             parents: undefined,
             selectionKey: undefined,
             target: undefined
         };
 
-        const tab = props.location && props.location.query && props.location.query.tab ? props.location.query.tab : SampleInsertPanelTabs.Grid;
-        if (selectTab && tab != SampleInsertPanelTabs.Grid)
+        const tab = props.location && props.location.query && props.location.query.tab ? props.location.query.tab : EntityInsertPanelTabs.Grid;
+        if (selectTab && tab != EntityInsertPanelTabs.Grid)
             this.props.selectStep(parseInt(tab));
 
         let { insertModel } = this.state;
 
         if (insertModel
-            && insertModel.getTargetSampleSetName() === queryParams.target
+            && insertModel.getTargetEntityTypeName() === queryParams.target
             && insertModel.selectionKey === queryParams.selectionKey
-            && insertModel.parents === queryParams.parents
+            && (insertModel.originalParents === queryParams.parents || !this.allowParents())
         )
             return;
 
-        insertModel = new SampleIdCreationModel({
-            parents: queryParams.parents,
-            initialSampleSet: queryParams.target,
+        insertModel = new EntityIdCreationModel({
+            originalParents: this.allowParents() ? queryParams.parents : undefined,
+            initialEntityType: queryParams.target,
             selectionKey: queryParams.selectionKey,
-            sampleCount: 0,
+            entityCount: 0,
+            entityDataType: props.entityDataType,
         });
 
-        initSampleSetInsert(insertModel)
+        const isSampleInsert = this.isSampleEntity();
+        let schemaQueries = Map<string, SchemaQuery>();
+        let currentSchema = isSampleInsert ? SCHEMAS.EXP_TABLES.SAMPLE_SETS :  SCHEMAS.EXP_TABLES.DATA_CLASSES;
```

Comment:
can currentSchema be props instead? This feels a bit hardcoded and is limiting use to sample and dataclass. Same with schemaQueries.